modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISJsonOnlineTest.java

```diff
@@ -442,6 +444,87 @@ public void testJSONArrayContainsFunction() throws Exception {
+
+            testDelegatedArrayContains(
+                    ff,
+                    fs,
+                    "jsonColumn",
+                    "/type",
+                    "OTHERS",
+                    feature -> assertEquals("jsontest.10", feature.getIdentifier().getID()),
+                    feature ->
+                            assertTrue(
+                                    feature.getAttribute("jsonColumn")
+                                            .toString()
+                                            .contains("\"type\": \"OTHERS\"")));
+
+            if (pgJsonTestSetup.supportJsonB) {
+                testDelegatedArrayContains(
+                        ff,
+                        fs,
+                        "jsonbColumn",
+                        "/type",
+                        "OTHERS",
+                        feature -> assertEquals("jsontest.10", feature.getIdentifier().getID()),
+                        feature ->
+                                assertTrue(
+                                        feature.getAttribute("jsonbColumn")
+                                                .toString()
+                                                .contains("\"type\": \"OTHERS\"")));
+            }
+        }
+    }
+
+    @Test
+    public void testJSONArrayContainsFunctionRootArrayNumber() throws Exception {
+        if (pgJsonTestSetup.supportJsonPathExists) {
+            ContentFeatureSource fs = dataStore.getFeatureSource(tname("jsontest"));
+            FilterFactory ff = dataStore.getFilterFactory();
+
+            testDelegatedArrayContains(
+                    ff,
+                    fs,
+                    "jsonColumn",
+                    "/version",
+                    3,
+                    feature -> assertEquals("jsontest.11", feature.getIdentifier().getID()),
+                    feature ->
+                            assertTrue(
+                                    feature.getAttribute("jsonColumn")
+                                            .toString()
+                                            .contains("\"version\": 3")));
+
+            if (pgJsonTestSetup.supportJsonB) {
+                testDelegatedArrayContains(
+                        ff,
+                        fs,
+                        "jsonbColumn",
+                        "/version",
+                        3,
+                        feature -> assertEquals("jsontest.11", feature.getIdentifier().getID()),
+                        feature ->
+                                assertTrue(
+                                        feature.getAttribute("jsonbColumn")
+                                                .toString()
+                                                .contains("\"version\": 3")));
+            }
+        }
+    }
+
+    @Test
+    public void testJSONArrayContainsWithJsonPathExists() throws Exception {
+        if (pgJsonTestSetup.supportJsonPathExists) {
```

Comment:
This if has no reason to exist, if methods invoked are also marked with @Test, they are going to be invoked directly anyways. If those tests rely on having json path exist support to work, then you should add an Assume at their beginning, so that they can be skipped if the assumption fails.

Reply:
removed